goethe/src/main/java/com/palantir/goethe/Goethe.java

```diff
@@ -244,19 +234,18 @@ public static Path formatAndEmit(String fullClassName, String source, Path baseD
      * e.g., {@code com.foo.bar.MyClass -> /<baseDir>/com/foo/bar/MyClass.java} and creates all directories.
      */
     private static Path getFilePath(Path baseDir, String packageName, String typeName) throws IOException {
+        String fullyQualifiedTypeName = packageName.isEmpty() ? typeName : packageName + '.' + typeName;
+        return getFilePath(baseDir, fullyQualifiedTypeName);
+    }
+
+    private static Path getFilePath(Path baseDir, String fullyQualifiedTypeName) throws IOException {
```

Comment:
nit: consistent names

```suggestion
private static Path getFilePath(Path baseDir, String className) throws IOException {
```